httomo/wrappers_class.py

```diff
@@ -0,0 +1,237 @@
+                    self.wrapper_method = super(TomoPyWrapper, self)._execute_normalize
+            if function_name == "algorithm":
+                self.wrapper_method = super(TomoPyWrapper, self)._execute_reconstruction
+            if function_name == "rotation":
+                self.wrapper_method = super(TomoPyWrapper, self)._execute_rotation
+
+
+# TODO: generilise HttomolibWrapper wrappers for GPU runs
+
+
+class HttomolibWrapper(BaseWrapper):
+    """A class that wraps httomolib functions for httomo"""
+
+    def __init__(
+        self, module_name: str, function_name: str, method_name: str, comm: Comm
+    ):
+        super().__init__(module_name, function_name, method_name, comm)
+
+        # if not changed bellow the generic wrapper will be executed
+        self.wrapper_method = super(HttomolibWrapper, self)._execute_generic
+
+        if module_name in ["misc", "prep", "recon"]:
+            from importlib import import_module
+
+            self.module = getattr(
+                import_module("httomolib." + module_name), function_name
+            )
+            if function_name == "images":
+                self.wrapper_method = self._execute_images
+            if function_name == "normalize":
+                func = getattr(self.module, method_name)
+                sig_params = signature(func).parameters
+                if "dark" in sig_params and "flat" in sig_params:
+                    self.wrapper_method = super(
+                        HttomolibWrapper, self
+                    )._execute_normalize
+            if function_name == "algorithm":
+                self.wrapper_method = super(
+                    HttomolibWrapper, self
+                )._execute_reconstruction
+            if function_name == "rotation":
+                self.wrapper_method = super(HttomolibWrapper, self)._execute_rotation
+
+    def _execute_images(
+        self, method_name: str, params: Dict, out_dir: str, comm: Comm, data: xp.ndarray
+    ) -> None:
+        """httomolib wrapper for save images function.
+
+        Args:
+            method_name (str): The name of the method to use.
+            params (Dict): A dict containing independent of httomo params.
+            out_dir (str): The output directory.
+            comm (Comm): The MPI communicator.
+            data (xp.ndarray): a numpy or cupy data array.
+
+        Returns:
+            None: returns None.
+        """
+        # as now this function does not require ncore parameter
+        # TODO: not elegant, needs rethinking
+        try:
+            del params["ncore"]
+        except:
+            pass
+
+        if gpu_enabled:
+            data = getattr(self.module, method_name)(
+                xp.asnumpy(data), out_dir, comm_rank=comm.rank, **params
+            )
+        else:
+            data = getattr(self.module, method_name)(
+                data, out_dir, comm_rank=comm.rank, **params
+            )
```

Comment:
```suggestion
        for key in ["ncore", ""]:  #: silently remove `ncore` param.
            params.pop(key, None)
        data = getattr(self.module, method_name)(
            data.get() if gpu_enabled else data, out_dir, comm_rank=comm.rank, **params
        )
```

Reply:
thanks, not quite sure that this does as I need to convert possibly a cupy array to a numpy one if the gpu is enabled.
```
data = getattr(self.module, method_name)(
            data.get() if gpu_enabled else data, out_dir, comm_rank=comm.rank, **params
        )
```

Reply:
`.get()` is a way to get a copy of the ndarray on host memory: https://docs.cupy.dev/en/stable/reference/generated/cupy.ndarray.html#cupy.ndarray.get

Reply:
Either will work!